### DISCUSSION TITLE
Watermelondb and performance tips

### DISCUSSION THREAD

**Discussion by mlecoq** [Nozbe/WatermelonDB#1157] [General]
Hi everyone, I have a screen with several subscription to display a list and within a modal with a form to create new elements (with its own subscriptions). When I submit the form (on Android), there is a small freeze on the screen. 

The freeze leads user to press again on submit button and 2 items are created instead of one. 

I am wondering if I can lighten observers. On this screen there is mostly creation. Have you any advice when we have to deal with many observations ?

**Comment by salembaira** [comment]
Hi,

You should maybe debounce the onclick button function. It will prevent double creation.

**Reply by mlecoq** [reply]
Hi, thanks for your answer, I have already tried to debounce without success. In fact I was able to solve this issue by replacing React Navigation Stack Navigator by Native Stack Navigator and rebuilding my form (replacing formik with react-hook-form to use FormContext)

**Reply by salembaira** [reply]
> Hi, thanks for your answer, I have already tried to debounce without success. In fact I was able to solve this issue by replacing React Navigation Stack Navigator by Native Stack Navigator and rebuilding my form (replacing formik with react-hook-form to use FormContext)

Good thing that you found a solution.
